cli/pkg/local/app.go

```diff
@@ -182,9 +184,12 @@ func NewApp(ctx context.Context, opts *AppOptions) (*App, error) {
 		"rill.download_row_limit": "0", // 0 means unlimited
 		"rill.stage_changes":      "false",
 	}
-	for k, v := range opts.Variables {
-		vars[k] = v
+	maps.Copy(vars, opts.Variables)
+	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return nil, err
 	}
+	maps.Copy(vars, envMap)
```

Comment:
**Optimization Issue**: The use of godotenv.Read to load environment variables from a file during the execution of NewApp introduces potential I/O blocking and inefficiency, especially if the .env file is large or the disk I/O is slow. This can degrade the performance of the application startup. <br> **Fix**: To optimize the loading of environment variables and reduce I/O blocking, consider asynchronously loading the .env file or using a non-blocking approach. Additionally, evaluate if environment variables can be loaded once at the application startup and reused, rather than reading from the file system repeatedly. <br> **Code Suggestion**: 
 ```
 -	envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
 -	if err != nil && !errors.Is(err, os.ErrNotExist) {
 -		return nil, err
 +	go func() {
 +		envMap, err := godotenv.Read(filepath.Join(projectPath, ".env"))
 +		if err != nil and !errors.Is(err, os.ErrNotExist) {
 +			log.Error("Failed to read .env file", err)
 +		}
 +		maps.Copy(vars, envMap)
 +	}()
 ```